commands/publish

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+if [ -z "$PYPI_USER" ]; then
+    echo '$PYPI_USER is not set.'
+    exit 1
+fi    
+
+if [ -z "$PYPI_PASSWORD" ]; then
+    echo '$PYPI_PASSWORD is not set.'
+    exit 1    
+fi
+
+changes=`git diff`
+
+if [ -n "${changes}" ]; then
+  echo "You have made changes to this branch that you have not committed."
+  exit 1
+fi
+
+staged_changes=`git diff --staged`
+
+if [ -n "${staged_changes}" ]; then
+  echo "You have staged changes to this branch that you have not committed."
+  exit 1
+fi
+
+tagged=`git log -1 --decorate | head -1 | grep 'tag:'`
+
+if [ -z "$tagged" ]; then
+  # We don't encourage people to tag it unless they have no changes pending.
+  echo "You can only publish a tagged commit."
+  exit 1
+fi
+
+read -p "Are you sure you want to publish $(poetry version) to PyPi? "
```

Comment:
If you run this directly as part of the Travis build (as currently specified), the `publish` step will hang because the script will wait for input here. I think if you want a script like this as a command line utility for your own use then that is fine, but you probably want to run line 43 in this file as part of the `publish` build step. You can make it so the `publish` step only runs on tagged builds.